### PR TITLE
merged get tasks routes into one

### PR DESCRIPTION
**Pull request by KarandeepLubana**
Completed in this PR:

- Merged routes getTaskByDay, getTaskByMonth and getTasks into one route

### CODE REVIEW

File: backend/controllers/taskController.js [hunk 1]
```diff
@@ -1,184 +1,167 @@
...
+      try {
+        startDate = new Date(year, month - 1, day);
+        endDate = new Date(year_end, month_end - 1, day_end + 1);
+      } catch (error) {
+        throw new Error("Invalid Date Input");
+      }
     }
-
-    // Check if its a valid date object
     const tasks = await Task.find({
```
Comment (hiimchrislim) [inline backend/controllers/taskController.js]: I would also place this in the `if statement` portion and then add an `else` to indicate that there's an Invalid Date Input (As of right now, I can theoretically write something like `http://localhost:5000/api/tasks?start=202203&end=123981239812398712398712397813102938102938012983` which then it should return Invalid Date Input

File: backend/controllers/taskController.js [hunk 2]
```diff
@@ -1,184 +1,167 @@
...
+  } else if (!startDateParam && !endDateParam) {
+    await Task.find({ user_id: userId })
+      .then((docs) => {
+        res.status(200).json(docs);
+      })
+      .catch((err) => {
+        res.status(500);
+        throw new Error(`Could not fetch doc ${err}`);
+      });
+  } else {
```
Comment (hiimchrislim) [inline backend/controllers/taskController.js:94]: The else from this portion can be moved to my earlier comment about the `if statement`

Comment (hiimchrislim) [reply backend/controllers/taskController.js:94]: Actually, this can be left here. This is fine

**Review by hiimchrislim — CHANGES_REQUESTED** [review]
Overall, it looks good! I left a comment to take a look at.
If you have any questions, please let me know

File: backend/controllers/taskController.js (hunk 1 above)
Comment (KarandeepLubana) [reply backend/controllers/taskController.js]: Fixed: Changes are reflected in recent commits

**Review by hiimchrislim — APPROVED** [review]
Looks great!

**Pull request merged**